wordpress-mobile/WordPress-iOS
# Features/enable signin flow

This PR implements the new sign in flow everywhere the LoginViewController was being presented.  The approach allows for switching between the old and new sign in flow based on a toggle, where the old LoginViewController is the default.
The implementation supports an A/B test courtesy of Optimizely. 

The changes include updates to the WPAuthTokenIssueSolver which, after chatting with @diegoreymendez, we probably don't need anymore.  We'll plan on removing the class in a follow up PR after this one is merged.

To test:
- Launch the app and sign out.  Take note of which version of the sign in flow you see.  Perform the following tests.
- Test logging into wpcom while not signed into a self-hosted site.
- Test logging into a self-hosted site while not signed in to wpcom.
- Test logging into a wpcom while signed in with a self-hosted site. 
- Test logging into a self-hosted site while signed into wpcom. 
- Sign into Optimizely and edit the variable that sets which version of the a/b test you see. Repeat the previous tests for the other version of the sign in flow. 

Needs review: @kwonye would you vett the A/B testing since you're familiar with Optimizely. @koke could I trouble you for a code review as well?  Thanks!!
